util/src/main/scala/testz/futureUtil.scala

```diff
@@ -34,6 +34,26 @@ object futureUtil {
 
   import scala.concurrent.{Future, ExecutionContext}
 
+  def orIterator[A](it: Iterator[Future[Boolean]])(ec: ExecutionContext): Future[Boolean] = {
+    def outer(acc: Boolean): Future[Boolean] = {
+      @scala.annotation.tailrec
+      def inner(acc: Boolean): Future[Boolean] = {
+        if (it.hasNext) {
+          val ne = it.next
+          if (ne.isCompleted) {
+            inner(acc || ne.value.get.get)
+          } else {
+            ne.flatMap(b => outer(acc || b))(ec)
+          }
+        } else {
+          Future.successful(acc)
+        }
+      }
+      inner(acc)
```

Comment:
oh I get it. this makes `inner` tail recursive in the `ne.isCompleted` branch.

Reply:
Right, this code is fucking abhorrent but this is what I need to do for fast code inside `Future`. 🗡